tests/lib/rules/func-names.js

```diff
@@ -127,6 +119,28 @@ ruleTester.run("func-names", rule, {
             parserOptions: { ecmaVersion: 6 }
         },
 
+        // export default
+        {
+            code: "export default function foo() {}",
+            options: ["always"],
+            parserOptions: { sourceType: "module", ecmaVersion: 6 }
+        },
+        {
+            code: "export default function foo() {}",
+            options: ["as-needed"],
+            parserOptions: { sourceType: "module", ecmaVersion: 6 }
+        },
+        {
+            code: "export default function foo() {}",
```

Comment:
How is this valid? The option is `"never"` but the function has a name.

Reply:
That would require more checks, `foo` is here a declaration and can be used from within the module. And even if that's not the case, I believe this would break some builds

Reply:
The reasoning is that, under option never, function f() is allowed, export function f() is allowed, so likewise export default function f() should be allowed.

I understand in some perspective this should be reported, so this case is controvesial. Not reporting in this case would make both sides happy.